providers/base/bin/wifi_client_test_netplan.py

```diff
@@ -315,12 +316,32 @@ def print_route_info():
     print()
 
 
+def _validate_gateway_ip(gateway):
+    if not gateway:
+        return ""
+    # Check if the gateway is a valid IP address
+    # Examples:
+    # 192.168.144.1 (TP-Link 123)
+    # 192.168.144.1
+    ip_part = gateway.split()[0]  # Get the first part before any space
+    return ip_part if __validate_ip(ip_part) else ""
+
+
+def __validate_ip(ip):
+    try:
+        ipaddress.ip_address(ip)
+        return True
+    except ValueError:
+        return False
+
+
 def get_gateway(interface, renderer):
-    gateway = None
     info = get_interface_info(interface, renderer)
-    gateway = info.get("gateway", None)
+    gateway = info.get("gateway")
```

Comment:
so that we can always split
```suggestion
    gateway = info.get("gateway") or ""
```